docs/discussion/Multi-Tenancy Support.md

```diff
@@ -0,0 +1,168 @@
+- **Azure**: Users can configure Microsoft Entra ID(formerly known as Azure AD) and integrate their AKS cluster with Azure RBAC to [grant access at namespace level](https://learn.microsoft.com/en-us/azure/aks/access-control-managed-azure-ad#apply-just-in-time-access-at-the-namespace-level).
+- **AWS**: Users can manage permissions across namespaces in EKS with IAM roles and RBAC. You can refer to this [reference](https://repost.aws/knowledge-center/eks-iam-permissions-namespaces) for details.
+- **GCP**: GCP utilize Google Groups and IAM roles to control namespace access. For further information, you can consult this [reference](https://cloud.google.com/kubernetes-engine/docs/best-practices/enterprise-multitenancy).
+
+## Data Isolation
+### Isolation Level
+Data isolation in Kubernetes can be categorized into 2 types:
+- **Hard Multi-Tenancy**: It implies a scenario in which tenants have no trust in each other. In this case, different tenants should not have access to namespaces owned by others. Furthermore, the configurations and operations of one tenant should not affect or make a difference to other namespaces.
+- **Soft Multi-Tenancy**: It fits the case when tenants could be trusted so that tenants could share a cluster within an organization.
+
+In terms of the Ratify use case, soft multi-tenancy would be more suitable since hard multi-tenancy usually requires a dedicated cluster for each tenant.
+
+### Control Plane Isolation
+To ensure control plane isolation, Ratify will implement namespace and Role-Based Access Control (RBAC), as detailed in the preceding section, to segregate access to API resources among different users.
+
+### Data Plane Isolation
+Firstly we need to consider what kind of data needs to be isolated or shared.
+#### Custom Resource
+Ratify has 4 CRDs defined, each of them is crucial to validation workflow. Different teams should have their own CRs generated and cannot alter other teams' CRs.
+#### Cache
+In the current configuration, Ratify offers support for both in-memory cache and distributed cache.
+Concerning in-memory cache, neither the cluster admin nor the team admin possesses direct access. The key focus here is on Ratify's ability to effectively partition the cache for distinct teams, mitigating concerns regarding cache isolation.
+In the case of distributed cache, exemplified by Redis, team admins and developers lack direct access to the Redis cluster unless explicitly granted permissions. Notably, cluster admins retain the capability to access Redis pods for manual data retrieval or updates—an occurrence often encountered during debugging tasks. It is crucial to acknowledge a specific scenario where granting a team admin access to Redis pods may inadvertently expose data from other teams within those pods, warranting careful consideration.
```

Comment:
For distributed caching, Ratify uses an encryption key stored in a secret to encrypt all content stored at rest in Redis instance. Beyond access to the Redis pod, we can restrict access to the encryption key secret to only the cluster admin which should help harden isolation cross tenant.

Reply:
Thanks for calling it out, I've added this option as well.